packages/aws-library/src/aws_library/s3/_client.py

```diff
@@ -470,6 +474,59 @@ async def copy_objects_recursively(
             limit=_MAX_CONCURRENT_COPY,
         )
 
+    async def get_object_file_stream(
+        self,
+        bucket_name: S3BucketName,
+        object_key: S3ObjectKey,
+        *,
+        chunk_size: int = DEFAULT_READ_CHUNK_SIZE,
+    ) -> StreamData:
+        """stream read an object in S3 chunk by chunk"""
+
+        # NOTE `download_fileobj` cannot be used to implement this because
+        # it will buffer the entire file in memory instead of reading it
+        # chunk by chunk
+
+        # below is a quick call
+        head_response = await self._client.head_object(
+            Bucket=bucket_name, Key=object_key
+        )
+        file_size = FileSize(head_response["ContentLength"])
+
+        async def _(progress_bar: ProgressBarData) -> FileStream:
```

Comment:
who initialize that progress bar and with which size?

Reply:
progress was removed